Gibe.Umbraco.Blog/Composing/IndexEventsComponent.cs

```diff
@@ -133,40 +133,67 @@ private void ExternalIndexTransformingIndexValues(object sender, IndexingItemEve
 
 		private Dictionary<string, IEnumerable<object>> AddAuthorFields(IndexingItemEventArgs e)
 		{
-			var authorId = e.ValueSet.GetSingleValue<int?>(ExamineFields.PostAuthor);
+			var authorIdString = e.ValueSet.GetSingleValue(ExamineFields.PostAuthor);
+			var authorName = string.Empty;
+			if(int.TryParse(authorIdString, out var authorId))
+			{
+				authorName = GetUserName(authorId);
+			}
+			else if (UdiParser.TryParse(authorIdString, out var authorUid))
+			{
+				authorName = GetAuthorName(authorUid);
+			}
 			Dictionary<string, IEnumerable<object>> values = new Dictionary<string, IEnumerable<object>>();
 
-			if (authorId.HasValue)
+			if (!string.IsNullOrEmpty(authorName))
 			{
-				values.Add(ExamineFields.PostAuthorName, new[] { GetUserName(authorId.Value).ToLower()});
+				values.Add(ExamineFields.PostAuthorName, new[] { authorName.ToLower()});
 			}
 
 			return values;
 		}
 
 		private Dictionary<string, IEnumerable<object>> AddTagFields(IndexingItemEventArgs e)
 		{
-			Dictionary<string, IEnumerable<object>> values = new Dictionary<string, IEnumerable<object>>();
-			var tagValue = e.ValueSet.GetSingleValue(ExamineFields.Tags);
-			if (tagValue != null)
+			var values = new Dictionary<string, IEnumerable<object>>();
+
+			var tagMultipleValues = e.ValueSet.GetValues(ExamineFields.Tags);
+			if (tagMultipleValues?.Any() == true)
 			{
-				try
+				AddTagsToIndex(tagMultipleValues);
+			}
+			else
+			{
+				var tagSingleValue = e.ValueSet.GetSingleValue(ExamineFields.Tags);
+				if (tagSingleValue != null)
 				{
-					var tags =
-						JsonConvert.DeserializeObject<IEnumerable<string>>(tagValue);
-					if (tags != null)
+					try
+					{
+						var tags = JsonConvert.DeserializeObject<IEnumerable<string>>(tagSingleValue);
+						if (tags != null)
+						{
+							AddTagsToIndex(tags);
+						}
+					}
+					catch (JsonReaderException)
 					{
-						values.Add(ExamineFields.Tag, tags.Select(t => t.ToLower()));
+						// Tags are likely csv rather than JSON
```

Comment:
Good question.  I think it explains what's going on cuz it might seem a bit weird that we try to deserialize this value, then do a string.Split() - makes it known this is intentional.

Don't feel strongly about it either way though